back-end/app.js

```diff
@@ -332,11 +333,32 @@ app.get('/shows-trakt/:id', (req, res, next) => {
           //add response from Trakt API to final response object
           response_final = responseA.data
           //return tmdb_id value which is used for retrieving poster url in get request to Tmdb API
-          return responseA.data.ids.tmdb
+          return responseA.data
+        })
+        .then((responseX) => {
+          //retrieve season info for a show from Seasons Trakt API
+          if (tmdbType=='tv'){
+            return axios.get(seasonURL, {
+              headers: {
+                'trakt-api-version': '2',
+                'trakt-api-key': `${process.env.API_KEY_TRAKT}`,
+                Accept: '*/*', //necessary since requests have multiple content-types
+                'User-Agent': 'request'
+              }
+            })
+          }
         })
         .then((responseB) => {
+          let count=0;
+          //show seasons only for shows
+          if (tmdbType=='tv'){
+            //count season objects
+            for (let i=0; i<responseB.data.length; i++)
+              count+=1
+            response_final['seasons']=count //set number of seasons
```

Comment:
Instead of going through this for loop to count up the seasons, could you maybe just set `response_final['seasons']` equals to `responseB.data.length` instead?